compiler/src/dotty/tools/repl/ReplCompiler.scala

```diff
@@ -146,7 +146,7 @@ class ReplCompiler extends Compiler {
     val objectTermName = ctx.source.file.toString.toTermName
     objectNames.update(defs.state.objectIndex, objectTermName)
 
-    val unit = CompilationUnit(ctx.source)
+    val unit = new ReplCompilationUnit(ctx.source)
```

Comment:
Nit: I would use an anonymous class
```suggestion
    val unit = new CompilationUnit(ctx.source) {
      override def isSuspendable: Boolean = false
    }
```

Reply:
I wanted to make it possible to identify a REPL compilation unit simply. I remember seeing places where we identify them by the name of the file which we could simplify with this scheme.